src/models/user.js

```diff
@@ -0,0 +1,29 @@
+const mongoose = require('mongoose');
+
+const {Schema} = mongoose;
+
+const userSchema = new Schema({
+    fullName: {
+        type: String,
+        required: true
+    },
+    email: {
+        type: String,
+        required: true
+    },
+    phoneNumber: {
+        type: String,
+        required: true
+    },
+    password: {
+        type: String,
+        required: true
+    },
+    promoCode: {
+        type: String,
+        required: false
+    }
+});
+
+const user = mongoose.model('user', userSchema);
+module.exports = user;
```

Comment:
You coud just do module.exports = mongoose.model('user', userSchema)

Reply:
Yes I agree, it'll make it more concise for line 29.

This one is equally a suggestion, what we have is not bad either, I was thinking you could require a destructured  Schema and model so we use them directly in the script.
Or call the Schema on the mongoose variable for line 3 like you have done for line 28.
so that is either
const { Schema, model } = require('mongoose');
or in line 5
const userSchema = new mongoose.Schema({...});
then line 3 goes